sklearn/svm/src/liblinear/linear.cpp

```diff
@@ -70,6 +79,52 @@ static void info(const char *fmt,...)
 static void info(const char *fmt,...) {}
 #endif
 
+// New random number generator replacing `rand()`, to ensure the same behaviour on windows-linux, with increased speed
+// - (1) Init a `mt_rand` object
+#if INT_MAX == 0x7FFFFFFF
+std::mt19937 mt_rand(std::mt19937::default_seed);
+#elif INT_MAX == 0x7FFFFFFFFFFFFFFF
+std::mt19937_64 mt_rand(std::mt19937::default_seed);
+#else
+info("Random number generator is not fixed for this system. Please report issue. INT_MAX=%d\n", INT_MAX);
+exit(1);
+#endif
+
+// - (2) New public `set_seed()` function that should be used instead of `srand()` to set a new seed.
+void set_seed(unsigned custom_seed) {
+    mt_rand.seed(custom_seed);
+}
+
+// - (3) New internal `bounded_rand_int` function, used instead of rand() everywhere.
+inline int bounded_rand_int(int orig_range) {
+    // "Legacy way" - make a 31bit or 63bit positive random number
```

Comment:
Again it's not explicit that by "legacy" you mean original liblinear code does it this way. But maybe it's clear enough.

Reply:
I'll improve the comment